azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -400,7 +462,7 @@ func resourceSubscriptionDelete(d *pluginsdk.ResourceData, meta interface{}) err
 		return fmt.Errorf("failed to cancel Subscription %q (Alias %q): %+v", subscriptionId, id.Name, err)
 	}
 
-	return nil
+	return resourceSubscriptionRead(d, meta)
```

Comment:
We only ever return nil at the end of a delete, returning read here will result in Terraform presenting an erroneous not-found error to the user on success.
```suggestion
	return nil
```